frontend/packages/knative-plugin/src/utils/create-channel-utils.ts

```diff
@@ -0,0 +1,144 @@
+};
+
+export const useChannelList = (namespace: string): ChannelListProps => {
+  const [accessData, setAccessData] = useSafetyFirst({ loaded: false, channelList: [] });
+  const channelResourcesList = getDynamicChannelModelRefs();
+  const accessList = [];
+  React.useEffect(() => {
+    _.forIn(channelResourcesList, (channelRef: string) => {
+      if (isGroupVersionKind(channelRef)) {
+        const [group] = getGroupVersionKind(channelRef) ?? [];
+        const { plural } = modelFor(channelRef) || {};
+        accessList.push(
+          checkAccess({
+            group,
+            resource: plural,
+            namespace,
+            verb: 'create',
+          }).then((result) => (result.status.allowed ? channelRef : {})),
+        );
+      }
+    });
+    Promise.all(accessList)
+      .then((results) => {
+        const channelList = results.reduce((acc, result) => {
+          return [...acc, result];
+        }, []);
+
+        setAccessData({ loaded: true, channelList });
+      })
+      // eslint-disable-next-line no-console
+      .catch((err) => console.warn(err.message));
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, []);
+
+  return accessData;
+};
+
+export const getCreateChannelData = (formData: AddChannelFormData): K8sResourceKind => {
+  const {
+    type,
+    name,
+    data,
+    application: { name: applicationName },
+    namespace,
+  } = formData;
+  if (!isGroupVersionKind(type)) {
+    return {};
+  }
+  const defaultLabel = getAppLabels(name, applicationName);
+  const [channelGroup, channelVersion, channelKind] = getGroupVersionKind(type);
+  const channelSpecData = data[channelKind.toLowerCase()];
+  const eventSourceResource: K8sResourceKind = {
+    apiVersion: `${channelGroup}/${channelVersion}`,
+    kind: channelKind,
+    metadata: {
+      name,
+      namespace,
+      labels: {
+        ...defaultLabel,
+      },
+      annotations: getCommonAnnotations(),
+    },
+    spec: {
+      ...(channelSpecData && channelSpecData),
+    },
+  };
+
+  return eventSourceResource;
+};
+
+export const getCreateChannelResource = (formData: AddChannelFormData) => {
+  switch (getChannelKind(formData.type)) {
+    case EventingChannelModel.kind:
+    case EventingIMCModel.kind:
+    case EventingKafkaChannelModel.kind:
+      return getCreateChannelData(formData);
+    default:
+      return safeLoad(formData.yamlData);
+  }
+};
+
+export const getChannelData = (kind: string) => {
+  const channelData = {
+    kafkachannel: {
+      numPartitions: 1,
+      replicationFactor: 1,
+    },
+  };
+  return channelData[kind];
+};
+
+export const useDefaultChannelConfiguration = (namespace: string): [string, boolean] => {
+  const CHANNEL_CONFIGMAP_NAME = 'default-ch-webhook';
+  const CHANNEL_CONFIGMAP_NAMESPACE = 'knative-eventing';
+
+  const [defaultChannel, defaultChannelLoaded] = useK8sGet<K8sResourceKind>(
+    ConfigMapModel,
+    CHANNEL_CONFIGMAP_NAME,
+    CHANNEL_CONFIGMAP_NAMESPACE,
+  );
```

Comment:
does it needs any special treatment for non-admin user or all can access it?

Reply:
No, even the consoledeveloper (selfProvisioner & view access) user will be able to retrieve this information.